src/provider/provider_level_zero.c

```diff
@@ -207,32 +217,41 @@ static void init_ze_global_state(void) {
 #else
     const char *lib_name = "libze_loader.so";
 #endif
-    // check if Level Zero shared library is already loaded
-    // we pass 0 as a handle to search the global symbol table
+    // The Level Zero shared library should be already loaded by the user
+    // of the Level Zero provider. UMF just want to reuse it
+    // and increase the reference count to the the Level Zero shared library.
+    void *lib_handle =
+        utils_open_library(lib_name, UMF_UTIL_OPEN_LIBRARY_NO_LOAD);
+    if (!lib_handle) {
+        LOG_ERR("Failed to open Level Zero shared library");
```

Comment:
This should be a fatal error as `Init_ze_global_state_failed == true` results in a fatal in `ze_memory_provider_initialize()`.

Reply:
Done